server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -177,6 +178,27 @@ public DocsStats docStats() {
         }
     }
 
+    /**
+     * Returns the file sizes for the current commit
+     */
+    public long getLastCommitSizeInBytes() {
+        SegmentInfos segmentInfos = getLastCommittedSegmentInfos();
+        long total = 0;
+        for (final SegmentCommitInfo commitInfo : segmentInfos) {
+            try {
+                total += commitInfo.sizeInBytes();
+            } catch (IOException err) {
+                logger.warn(
+                    "Failed to read file size for shard: [{}], id: [{}], err: [{}]",
```

Comment:
If we have an unexpected failure, should we emit a value at all? I'm wondering if a partial value + log message could result in botched billing that gets overlooked, vs a complete stoppage in metrics would get alarmed on quickly (and more likely to be caught in testing).

Reply:
This would be a failure to read a segment file.  I'm wondering if we should indicate partial failures in this method and report those upstream?